array.go

```diff
@@ -861,3 +861,29 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+// Every runs the passed function on all the Elements
+// in the array and returns a new Array instance.
+//
+// Example:
+//
+//	array := NewArray(t, []interface{}{"foo", "bar"})
+//	newArray := array.Every(func(idx int, val *httpexpect.Value) {
+//		value.String().NotEmpty()
+//	})
+func (a *Array) Every(fn func(index int, value *Value)) *Array {
```

Comment:
* Before any other checks, `Every` should check if the chain is failed already, and return in this case.

* The implementation will crash if `fn` is nil. Instead, if it's nil, we should report failure (AssertUsage) and return. See how we do it in Request.WithMatcher, for example.

   To make the assertion path in failure meaningful, you'll need to do it like this:

   ```go
   if fn == nil {
     a.chain.enter("Every()")
     defer a.chain.leave()
     a.chain.fail(...)
   }
   ```

Reply:
Also I suggest to move `Every` upper in file: right before `Empty` method (for both array and objects).

Reply:
Sure thing.